src/sdk/namespace_fs.js

```diff
@@ -2772,6 +2772,7 @@ class NamespaceFS {
      * @returns {Promise<string>}
      */
     async _find_version_path(fs_context, { key, version_id }, return_md_path) {
+        key = await this._get_correct_key_path(fs_context, key);
```

Comment:
can you explain this change please?

Reply:
@romayalon if the key is a partial dir path such as `a/b/c` this method will check whether the key is dir or not if its dir, then a backslash is added to the key to avoid the `AccessDeniedError` error. file path will leave key as it is

Reply:
But notice that we expect to fail with ENOENT when key doesn't end with a slash, and the folder is not a content dir - 
```
if (!stat.xattr?.[XATTR_DIR_CONTENT] || !params.key.endsWith('/')) {
throw error_utils.new_error_code('ENOENT', 'NoSuchKey');
```
if you change the key to have a / at the end, will we fail on this situation as we should per AWS?